### PR TITLE
Netplay: don’t show traversal server/port options (INI config only)

### PR DESCRIPTION
**Pull request by mathieui**
It's confusing for users, and websites or people willing to set up one can provide steps to edit config files, or pre-made config files.

### CODE REVIEW

File: Source/Core/DolphinWX/NetPlay/NetPlaySetupFrame.cpp [hunk 1]
```diff
@@ -246,9 +230,20 @@ void NetPlaySetupFrame::MakeNetPlayDiag(int port, const std::string &game, bool
 	else
 		trav = false;
 
-	unsigned long centralPort = 0;
-	m_traversal_port->GetValue().ToULong(&centralPort);
-	netplay_client = new NetPlayClient(ip, (u16)port, npd, WxStrToStr(m_nickname_text->GetValue()), trav, WxStrToStr(m_traversal_server->GetValue()), (u16)centralPort);
+	IniFile inifile;
+	const std::string dolphin_ini = File::GetUserPath(D_CONFIG_IDX) + "Dolphin.ini";
+	inifile.Load(dolphin_ini);
+	IniFile::Section& netplay_section = *inifile.GetOrCreateSection("NetPlay");
```
Comment (lioncash) [inline Source/Core/DolphinWX/NetPlay/NetPlaySetupFrame.cpp]:
```cpp
const std::string dolphin_ini = File::GetUserPath(D_CONFIG_IDX) + "Dolphin.ini";
IniFile inifile;
inifile.Load(dolphin_ini);
IniFile::Section& netplay_section = *inifile.GetOrCreateSection("NetPlay");
```

File: Source/Core/DolphinWX/NetPlay/NetPlaySetupFrame.cpp [hunk 2]
```diff
@@ -288,9 +283,20 @@ void NetPlaySetupFrame::OnHost(wxCommandEvent&)
 	unsigned long port = 0;
 	m_host_port_text->GetValue().ToULong(&port);
 
-	unsigned long centralPort = 0;
-	m_traversal_port->GetValue().ToULong(&centralPort);
-	netplay_server = new NetPlayServer((u16)port, trav, WxStrToStr(m_traversal_server->GetValue()), (u16)centralPort);
+	IniFile inifile;
+	const std::string dolphin_ini = File::GetUserPath(D_CONFIG_IDX) + "Dolphin.ini";
+	inifile.Load(dolphin_ini);
+	IniFile::Section& netplay_section = *inifile.GetOrCreateSection("NetPlay");
```
Comment (lioncash) [inline Source/Core/DolphinWX/NetPlay/NetPlaySetupFrame.cpp]:
```cpp
const std::string dolphin_ini = File::GetUserPath(D_CONFIG_IDX) + "Dolphin.ini";
IniFile inifile;
inifile.Load(dolphin_ini);
IniFile::Section& netplay_section = *inifile.GetOrCreateSection("NetPlay");
```

**Comment by comex** [comment]
I have a separate patch in the works that does this plus some other UI changes, but this works, I guess.

**Comment by comex** [comment]
I think people who used the old version and had it save the blank server/port fields to their config won't see the new default.  Haven't tested.  Am I wrong?

**Comment by mathieui** [comment]
@comex That's right, I was thinking about it, and I from quick testing it appears to be the case

**Comment by Jofzar** [comment]
I agree with this change, its silly how many people ask how to fix there netplay when they have accidentaly deleted the text, my other suggestion would be having a drop down box with a list a servers, this would allow for no confusion + the ability for custom servers for those websites which need them.

**Comment by mathieui** [comment]
This new commits makes it work if the option is empty or only contains spaces, but I'm not sure how to handle people who filled this entry incorrectly while they could edit it from the UI

**Comment by Jofzar** [comment]
The menus kind of look a bit empty now, maybe move nickname menu to the right with the box justified to the right?

http://i.imgur.com/iYDNtUV.png

(crappy paint job)

As for the problem with people who have incorrectly edited it, maybe make a default button, where it will reset it, and then add a tip to the netplay options about if it cant connect to try defaulting the server.

**Pull request merged**